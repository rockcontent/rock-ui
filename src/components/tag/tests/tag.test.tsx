import * as React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Tag, TagCloseButton, TagLabel } from '../';

describe('tag component tests', () => {
  it('Tag (only) renders correctly', () => {
    const { asFragment } = render(<Tag>A</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Tag (only) renders correctly', () => {
    const { getByTestId } = render(<Tag data-testid="tag">A</Tag>);
    expect(getByTestId('tag')).toBeInTheDocument();
  });

  it('Tag with close button renders correctly', () => {
    const { getByTestId } = render(
      <Tag data-testid="tag">
        <TagLabel data-testid="label">B</TagLabel>
        <TagCloseButton aria-label="Close Button" data-testid="close-btn" />
      </Tag>
    );
    expect(getByTestId('tag')).toBeInTheDocument();
    expect(getByTestId('label')).toBeInTheDocument();
    expect(getByTestId('close-btn')).toBeInTheDocument();
    expect(getByTestId('close-btn')).toHaveAttribute('aria-label');
  });
});
