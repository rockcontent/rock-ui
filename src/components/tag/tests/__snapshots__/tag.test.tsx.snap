// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag component tests Tag (only) renders correctly 1`] = `
<DocumentFragment>
  <span
    class="css-1qm7lvz"
  >
    A
  </span>
</DocumentFragment>
`;
